File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    dic = {}
    dic[words[0]] = True
    
    for i in range(1, len(words)):
        before, now = words[i-1], words[i]
        
        if now in dic or before[-1] != now[0]:
            return [(i % n) + 1, i // n + 1]
        
        dic[now] = True
    
    return [0, 0]

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    const dict = {};
    dict[words[0]] = true;
    
    for (let i = 1; i < words.length; i++) {
        const [before, now] = [words[i-1], words[i]];
        
        if (dict[now] || before.charAt(before.length-1) !== now.charAt(0)) {
            return [(i % n) + 1, Math.floor(i / n) + 1];
        }
        
        dict[now] = true;
    }
    
    return [0, 0];
}
